scipy/datasets/__init__.py

```diff
@@ -12,6 +12,47 @@
    face - Get example image for processing
    electrocardiogram - Load an example of a one-dimensional signal.
 
+
+Usage of Datasets
+^^^^^^^^^^^^^^^^^
+
+SciPy dataset files are stored within individual github repositories under
+the SciPy Org, following a naming convention as ``'dataset-<name>'``,
+for example `scipy.datasets.face` files live at
+https://github.com/scipy/dataset-face.
+The `scipy.datasets` submodule utilizes and depends on
+`Pooch <https://www.fatiando.org/pooch/latest/>`_, a python package built to
+simplify fetching data files. Pooch uses these repos to retrieve
+the respective dataset files when calling the dataset function.
+
+A registry of all the datasets, essentially a mapping of filenames
+with their SHA256 hash and repo urls are maintained,
+which pooch uses to handle and verify the downloads
+on function call. After downloading the dataset once, the files
+are saved in the system cache directory under ``'scipy-data'``.
+
+Dataset cache locations may vary on different platforms.
+
+For Mac::
+
+    '~/Library/Caches/scipy-data'
+
+For Unix::
+
+    '~/.cache/scipy-data' or the value of the XDG_CACHE_HOME environment
+    variable, if defined.
```

Comment:
I tested this, and imho it looks better here than for scikit-image:
```
$ ls ~/.cache/sci
scikit-image/ scipy-data/

$ ls ~/.cache/scipy-data/
ascent.dat  ecg.dat  face.dat

$ ls ~/.cache/scikit-image/
0.18.1
$ ls ~/.cache/scikit-image/0.18.1/data/
astronaut.png                      coffee.png                         lbpcascade_frontalface_opencv.xml  page.png
brick.png                          coins.png                          lfw_subset.npy                     phantom.png
camera.png                         color.png                          logo.png                           README.txt
cell.png                           grass.png                          microaneurysms.png                 retina.jpg
chelsea.png                        gravel.png                         moon.png                           rocket.jpg
chessboard_GRAY.png                horse.png                          motorcycle_disp.npz                text.png
chessboard_RGB.png                 hubble_deep_field.jpg              motorcycle_left.png                
clock_motion.png                   ihc.png                            motorcycle_right.png
```

This means that for scikit-image you get multiple copies on your machine, because it includes the version number. Data files should not be changing; if they do then any conflict can be resolved by either not reusing the same name or by clearing the cache (should work, unless there's a dramatic change). I don't think we have ever renamed a data file though.